applications/GeoMechanicsApplication/custom_conditions/T_microclimate_flux_condition.cpp

```diff
@@ -0,0 +1,483 @@
+{
+    double result = 0.0;
+
+    for (unsigned int i = 0; i < TNumNodes; ++i)
+    {
+        const auto net_radiation = CalculateNetRadiation(i);
+        result += net_radiation / TNumNodes;
+    }
+
+    return result;
+}
+
+template <unsigned int TDim, unsigned int TNumNodes>
+double TMicroClimateFluxCondition<TDim, TNumNodes>::CalculateNetRadiation(unsigned int NodeIndex) const
+{
+    auto& r_geom = this->GetGeometry();
+
+    // Eq 5.16
+    const auto incoming_radiation =
+        r_geom[NodeIndex].FastGetSolutionStepValue(SOLAR_RADIATION);
+    const auto short_wave_radiation = (1.0 - mAlbedoCoefficient) * incoming_radiation;
+
+    // Eq 5.17
+    const auto atmospheric_temperature =
+        r_geom[NodeIndex].FastGetSolutionStepValue(AIR_TEMPERATURE);
+    const auto absorbed_long_wave_radiation =
+        EffectiveEmissivity * BoltzmannCoefficient *
+        std::pow(atmospheric_temperature + 273.15, 4.0);
+
+    // Eq 5.18
+    const auto initial_soil_temperature =
+        r_geom[NodeIndex].FastGetSolutionStepValue(TEMPERATURE, 1);
+    const auto emitted_long_wave_radiation =
+        BoltzmannCoefficient * std::pow(initial_soil_temperature + 273.15, 4.0);
+
+    // Eq 5.15
+    return short_wave_radiation + absorbed_long_wave_radiation - emitted_long_wave_radiation;
+}
+
+template <unsigned int TDim, unsigned int TNumNodes>
+double TMicroClimateFluxCondition<TDim, TNumNodes>::CalculateSurfaceHeatStorage(
+    double TimeStepSize, double PreviousRadiation, double NetRadiation) const
+{
+    // Eq 5.20
+    return mFirstCoverStorageCoefficient * NetRadiation +
+           mSecondCoverStorageCoefficient * (NetRadiation - PreviousRadiation) / TimeStepSize +
+           mThirdCoverStorageCoefficient;
+}
+
+template <unsigned int TDim, unsigned int TNumNodes>
+WaterFluxes TMicroClimateFluxCondition<TDim, TNumNodes>::CalculateWaterFluxes(
+    unsigned int NodeIndex, double TimeStepSize, double PreviousStorage, double NetRadiation, double SurfaceHeatStorage) const
+{
+    const auto potential_evaporation =
+        CalculatePotentialEvaporation(NodeIndex, NetRadiation, SurfaceHeatStorage);
+
+    auto& r_geom = this->GetGeometry();
+    const auto precipitation = r_geom[NodeIndex].FastGetSolutionStepValue(PRECIPITATION);
+    const auto potential_storage =
+        PreviousStorage + TimeStepSize * (precipitation - potential_evaporation);
+
+    auto actual_precipitation = precipitation;
+    auto actual_evaporation = potential_evaporation;
+
+    // Correct the precipitation and evaporation if the storage is out of bounds
+    if (potential_storage > mMaximalStorage)
+    {
+        actual_precipitation =
+            (mMaximalStorage - PreviousStorage) / TimeStepSize + actual_evaporation;
+    }
+    else if (potential_storage < mMinimalStorage)
+    {
+        actual_evaporation = (PreviousStorage - mMinimalStorage) / TimeStepSize + actual_precipitation;
+    }
```

Comment:
I have taken some time to try and see if this is possible, but at this moment, I really don't see how. The problem is that the if-statements will remain necessary, because depending on if they're true, we either correct precipitation or evaporation. And the statements themselves are also not trivial to put into std::min/max format. If you have a good suggestion, please let me know! Or maybe @avdg81 sees a possibility?

Reply:
No, I also don't see how we could do that. Therefore, I'd suggest to keep it the way it is now. Unless @WPK4FEM has an elegant solution.

Reply:
I do not see an easy way. Please ignore my initial remark.